### PR TITLE
113_calculate_cost

### PR DESCRIPTION
**Pull request by Dmac26**
﻿## Summary
Addresses issue:
https://app.zenhub.com/workspaces/open-forest-58f8bbe105a35fad2d275a0c/issues/usdaforestservice/usfs-timber-permitting/113

This PR address all aspects of the 113 card, making updates to both the server side and frontend to allow users to input the number of cords they wish to harvest, while simultaneously restricting the amount they can enter to a range specified by each forest and also updating a display of the "totalCost" for a user's application.

## This pull request is ready to merge when...
- [x] Feature branch starts with the issue number
- [x] Is connected to its original issue via zenhub 👇
- [x] All tests are passing and meet coverage, linting, and accessibility requirements. And no security vulnerabilities ⚫️(Circle)
- [ ] This code has been reviewed by someone other than the original author﻿## Summary

### CODE REVIEW

**Comment by Dmac26** [comment]
@mtlaney or @briandavidson - This was originally reviewed by Greg on a different PR (https://github.com/USDAForestService/fs-open-forest-platform/pull/1439). Abdul asked me to create a new PR in an an attempt to pass all tests. I made quite a few updates to this one though.

File: frontend/src/app/application-forms/validators/number-of-cords-validation.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { FormControl, ValidatorFn } from '@angular/forms';
+
+export function numberOfCordsValidator(num, minNum = -99999): ValidatorFn {
+  return (control: FormControl): { [key: string]: any } => {
+    const val = control.value;
+    if (val) {
+      const phoneNumberRegex = /^(\d+)$/;
+      const valid = phoneNumberRegex.test(val);
+      if (val < minNum) {
```
Comment (mtlaney) [inline frontend/src/app/application-forms/validators/number-of-cords-validation.ts:9]: I'm assuming we can remove the phoneregex and valid phoneregex in this function

File: server/dba/seeders/forests.js [hunk 2]
```diff
@@ -10,7 +10,9 @@ const forests = [
       org_structure_code: '11-01-04',
       description: 'Idaho Panhandle | Idaho',
       forest_abbr: 'ipnf',
-      wood_cost: 5,
+      wood_cost: 5.00,
+      min_cords: 4,
+      max_cords: 2,
       timezone: 'America/Boise',
```
Comment (mtlaney) [inline server/dba/seeders/forests.js:16]: it looks like ipnf's min and max are swapped here?

**Review by mtlaney — CHANGES_REQUESTED** [review]
See comments, tested locally though and is working with the exception of IPNF

File: frontend/src/app/application-forms/validators/number-of-cords-validation.ts (hunk 1 above)
Comment (Dmac26) [reply frontend/src/app/application-forms/validators/number-of-cords-validation.ts:9]: @mtlaney - ah yeah. I changed the REGEX but forgot to change the name. I updated that just now.

File: server/dba/seeders/forests.js (hunk 2 above)
Comment (Dmac26) [reply server/dba/seeders/forests.js:16]: @mtlaney - Another typo here. `max_cords` is supposed to be 12 here

**Review by mtlaney — APPROVED** [review]

File: server/dba/migrations/42-forests.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ module.exports = {
       endDate: { type: Sequelize.DATE, field: 'end_date' },
       contact: { type: Sequelize.STRING(500), field: 'contact' },
       mapLinks: { type: Sequelize.STRING(500), field: 'map_links' },
-      woodCost: { type: Sequelize.NUMERIC(8, 2), field: 'wood_cost' },
+      woodCost: { type: Sequelize.DOUBLE, field: 'wood_cost' },
```
Comment (briandavidson) [inline server/dba/migrations/42-forests.js:18]: Just curious why we're using a `DOUBLE` here? We don't use a `DOUBLE` for any other data type in the application. (Same for line 24 below)

File: server/dba/migrations/46-addForestCords.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+'use strict';
```
Comment (briandavidson) [inline server/dba/migrations/46-addForestCords.js:1]: Can you please apply proper indentation to this whole file?

File: server/dba/migrations/47-addPermitCords.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+'use strict';
```
Comment (briandavidson) [inline server/dba/migrations/47-addPermitCords.js:1]: This whole file needs indentation / formatting.

**Review by briandavidson — CHANGES_REQUESTED** [review]

File: server/dba/migrations/42-forests.js (hunk 3 above)
Comment (Dmac26) [reply server/dba/migrations/42-forests.js]: @briandavidson -  had so much trouble getting those migration files to work, I was trying everything. But, I switched em back to NUMBERIC  and pushed it up.

File: server/dba/migrations/46-addForestCords.js (hunk 4 above)
Comment (Dmac26) [reply server/dba/migrations/46-addForestCords.js:1]: @briandavidson - applied

File: server/dba/migrations/47-addPermitCords.js (hunk 5 above)
Comment (Dmac26) [reply server/dba/migrations/47-addPermitCords.js:1]: @briandavidson - I formatted this one as well

**Review by briandavidson — CHANGES_REQUESTED** [review]
Let's either update this input so that it's not possible to enter text characters (right now the validation only occurs after the field loses focus), or update the display so that there is some sort of protection against showing the user something like this:

image

**Review by briandavidson — APPROVED** [review]
This is looking solid now, nice work 💥

**Pull request merged**